compiler/modules/comm.diagnostic/diagnostic.bal

```diff
@@ -27,6 +27,7 @@ public type File readonly & object {
     public function directory() returns string?;
     public function lineColumn(Position pos) returns LineColumn;
     public function lineContent(Position pos) returns string;
+    public function tokenLength(Position startPos) returns int;
```

Comment:
A better way to approach this is to change `lineContent` to:

```
public function lineContent(Position|Range range) returns string[3];
```

This returns the line containing the start of the range split into 3 segments:
1. The segment before the range
2. The segment containing the range (where the range is a token when the argument is a Position)
3. The segment after the range

This will enable us to deal with tabs properly (which we don't currently): generate what you call the `padding` string by replacing every character _other than tab_ by a space character in the first segment.

Reply:
fixed